### PR TITLE
Make media library php71 style example

### PR DESCRIPTION
**Pull request by carakas**
## Type

- Enhancement

## Pull request description

upgrade the media library and galleries to php 7.1

### CODE REVIEW

File: src/Backend/Modules/MediaLibrary/Component/StorageProvider/LocalStorageProvider.php [hunk 1]
```diff
@@ -107,13 +73,14 @@ public function getWebPath(MediaItem $mediaItem): string
...
+     *
      * @return string
      */
-    public function getWebPathWithFilter(MediaItem $mediaItem, string $filter): string
+    public function getWebPathWithFilter(MediaItem $mediaItem, string $filter = null): string
     {
         $webPath = $this->getWebPath($mediaItem);
 
-        if (!$mediaItem->getType()->isImage()) {
+        if ($filter === null && !$mediaItem->getType()->isImage()) {
```
Comment (jeroendesloovere) [inline src/Backend/Modules/MediaLibrary/Component/StorageProvider/LocalStorageProvider.php]: This should be an || comparison.

File: src/Backend/Modules/MediaLibrary/Component/StorageProvider/LocalStorageProvider.php [hunk 2]
```diff
@@ -107,13 +73,14 @@ public function getWebPath(MediaItem $mediaItem): string
     /**
      * @param MediaItem $mediaItem
      * @param string|null $filter The LiipImagineBundle filter name you want to use.
+     *
      * @return string
      */
-    public function getWebPathWithFilter(MediaItem $mediaItem, string $filter): string
+    public function getWebPathWithFilter(MediaItem $mediaItem, string $filter = null): string
```
Comment (jeroendesloovere) [inline src/Backend/Modules/MediaLibrary/Component/StorageProvider/LocalStorageProvider.php]: Maby rename $filter to $liipImagineBundleFilter and remove the PHPDoc.

File: src/Backend/Modules/MediaLibrary/Domain/MediaGroupMediaItem/MediaGroupMediaItem.php [hunk 3]
```diff
@@ -86,27 +81,18 @@ private function __construct(
...
-     * @param \Backend\Modules\MediaLibrary\Domain\MediaItem\MediaItem $item
-     * @param $sequence
-     * @return MediaGroupMediaItem
-     */
     public static function create(
         MediaGroup $group,
         MediaItem $item,
         int $sequence
-    ) : MediaGroupMediaItem {
+    ) : self {
```
Comment (jeroendesloovere) [inline src/Backend/Modules/MediaLibrary/Domain/MediaGroupMediaItem/MediaGroupMediaItem.php]: `): self`

File: src/Backend/Modules/MediaLibrary/Domain/MediaItem/MediaItemDataGrid.php [hunk 4]
```diff
@@ -85,31 +67,17 @@ private function getColumnsThatNeedToBeHidden(Type $type): array
...
     {
         return (string) (new self($type, $folderId))->getContent();
     }
 
-    /**
-     * @param Type $type
-     * @return \SpoonFormDropdown
-     */
-    private function getMassActionDropdown(Type $type)
+    private function getMassActionDropdown(Type $type): \SpoonFormDropdown
```
Comment (jeroendesloovere) [inline src/Backend/Modules/MediaLibrary/Domain/MediaItem/MediaItemDataGrid.php]: I would import \SpoonFormDropdown

File: src/Backend/Modules/MediaLibrary/Domain/MediaItem/MediaItemDataGrid.php [hunk 5]
```diff
@@ -85,31 +67,17 @@ private function getColumnsThatNeedToBeHidden(Type $type): array
...
     {
         $ddmMediaItemMassAction = new \SpoonFormDropdown(
```
Comment (jeroendesloovere) [inline src/Backend/Modules/MediaLibrary/Domain/MediaItem/MediaItemDataGrid.php]: Import as described in previous comment

File: src/Backend/Modules/MediaLibrary/Component/StorageProvider/LocalStorageProvider.php (hunk 2 above)
Comment (carakas) [inline src/Backend/Modules/MediaLibrary/Component/StorageProvider/LocalStorageProvider.php]: fixed in 97dd645

File: src/Backend/Modules/MediaLibrary/Component/StorageProvider/LocalStorageProvider.php (hunk 1 above)
Comment (jeroendesloovere) [inline src/Backend/Modules/MediaLibrary/Component/StorageProvider/LocalStorageProvider.php]: @carakas, don't forget the above comment.

Comment (carakas) [inline src/Backend/Modules/MediaLibrary/Component/StorageProvider/LocalStorageProvider.php]: fixed in 4fceb1764

Comment (carakas) [inline src/Backend/Modules/MediaLibrary/Component/StorageProvider/LocalStorageProvider.php]: fixed in 4fceb1764

File: src/Backend/Modules/MediaLibrary/Domain/MediaGroupMediaItem/MediaGroupMediaItem.php (hunk 3 above)
Comment (carakas) [inline src/Backend/Modules/MediaLibrary/Domain/MediaGroupMediaItem/MediaGroupMediaItem.php]: fixed in 1462e27c1

File: src/Backend/Modules/MediaLibrary/Domain/MediaItem/MediaItemDataGrid.php (hunk 4 above)
Comment (carakas) [inline src/Backend/Modules/MediaLibrary/Domain/MediaItem/MediaItemDataGrid.php]: fixed in 40bfa2a90

File: src/Backend/Modules/MediaLibrary/Domain/MediaItem/MediaItemDataGrid.php (hunk 5 above)
Comment (carakas) [inline src/Backend/Modules/MediaLibrary/Domain/MediaItem/MediaItemDataGrid.php]: fixed in 40bfa2a90

**Pull request merged**